### PR TITLE
Add ability to disable log in environments

### PR DESCRIPTION
**Pull request by altro3**
Need it to disable SLF4J initialization when micronaut environments used in micronaut-openapi.

ALso fixed some doclint errors and minor corections for slf4j.

### CODE REVIEW

**Comment by altro3** [comment]
@graemerocher @sdelamo @dstepanov With this feature, I can stop using slf4j-nop in micronaut-openapi

File: .gitattributes [hunk 1]
```diff
@@ -1,18 +1,19 @@
 # Auto detect text files and perform LF normalization
 *        text=auto
 
-*.java   text
-*.html   text
-*.kt     text
-*.kts    text
-*.md     text diff=markdown
+*.java   text eol=lf
```
Comment (sdelamo) [inline .gitattributes]: this changes seem unrelated to this pr.

**Comment by sdelamo** [comment]
> Need it to disable SLF4J initialization when micronaut environments used in micronaut-openapi.

can you elaborate what is the issue? I don't understand.

**Comment by altro3** [comment]
@sdelamo 

I use `DefaultEnvironment` class to load `application.yml` file from annotation processor micronaut-openapi, but this class use slf4j and when you run annotation processing you see this:
```
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
warning: unknown enum constant When.MAYBE
```

I've added `slf4j-nop` to solve this problem (https://github.com/micronaut-projects/micronaut-openapi/issues/821)

But, many people don't like this solution: https://github.com/micronaut-projects/micronaut-openapi/issues/908
And also this solution very bad for groovy projects: https://github.com/micronaut-projects/micronaut-openapi/issues/942

That's why I need to disable slf4j for `micronaut-openapi` and after that I can remove slf4j-nop from dependecies and don't scarry about slf4j initialization in openapi annotation level processing.

**Comment by altro3** [comment]
@sdelamo @graemerocher What do you think about this simple solution for slf4j problem?

**Review by graemerocher — APPROVED** [review]

File: .gitattributes (hunk 1 above)
Comment (sdelamo) [reply .gitattributes]: why do you need this?

Comment (altro3) [reply .gitattributes]: @sdelamo I can't commit my changes without this, otherwise build will be failed.

Now there is such a problem: if I leave this file as it is, then locally I cannot build the project. If I run SpotlesApply the project is built locally but not built on your СI-СD. My settings fix this problem, but it's time for you to fix the template itself so that I and other Windows users don't have these problems in the future.

Comment (sdelamo) [reply .gitattributes]: can you send a PR to [micronaut-project-template](https://github.com/micronaut-projects/micronaut-project-template/blob/master/.gitattributes) explain why this change is necessary?

Comment (sdelamo) [reply .gitattributes]: > I can't commit my changes without this, otherwise build will be failed.
> 
> Now there is such a problem: if I leave this file as it is, then locally I cannot build the project. If I run SpotlesApply the project is built locally but not built on your СI-СD. My settings fix this problem, but it's time for you to fix the template itself so that I and other Windows users don't have these problems in the future.

Is the issue that current `.gitattributes` is incompatible for Windows users? I don't understand what you say about CI. Currently, CI builds fine with our current `.gitattributes`

Comment (altro3) [reply .gitattributes]: @sdelamo done: https://github.com/micronaut-projects/micronaut-project-template/pull/348

File: inject/src/main/java/io/micronaut/context/env/AbstractPropertySourceLoader.java [hunk 2]
```diff
@@ -152,4 +157,22 @@ protected void processMap(Map<String, Object> finalMap, Map map, String prefix)
             }
         }
     }
+
+    /**
+     * Return logEnabled value.
+     *
+     * @return is log enabled
```
Comment (alvarosanchez) [inline inject/src/main/java/io/micronaut/context/env/AbstractPropertySourceLoader.java:164]:
```suggestion
     * @return is log enabled
     * @since 3.9.0
```

Same in `PropertySourcePropertyResolver`

File: inject/src/main/java/io/micronaut/context/env/AbstractPropertySourceLoader.java [hunk 3]
```diff
@@ -152,4 +157,22 @@ protected void processMap(Map<String, Object> finalMap, Map map, String prefix)
...
+     */
+    public boolean isLogEnabled() {
+        return logEnabled;
+    }
+
+    /**
+     * Setter for logEnabled.
+     *
+     * @param logEnabled is log enabled
```
Comment (alvarosanchez) [inline inject/src/main/java/io/micronaut/context/env/AbstractPropertySourceLoader.java:175]:
```suggestion
     * @param logEnabled is log enabled
     * @since 3.9.0    
```

Same in `PropertySourcePropertyResolver`

File: inject/src/main/java/io/micronaut/context/env/AbstractPropertySourceLoader.java [hunk 4]
```diff
@@ -95,17 +100,17 @@ public int getOrder() {
     private Map<String, Object> loadProperties(ResourceLoader resourceLoader, String qualifiedName, String fileName) {
         Optional<InputStream> config = readInput(resourceLoader, fileName);
         if (config.isPresent()) {
-            if (LOG.isDebugEnabled()) {
-                LOG.debug("Found PropertySource for file name: " + fileName);
+            if (logEnabled) {
```
Comment (alvarosanchez) [inline inject/src/main/java/io/micronaut/context/env/AbstractPropertySourceLoader.java:103]: Shouldn't this be:

```suggestion
            if (logEnabled && LOG.isDebugEnabled()) {
```

Same in the other places

**Review by alvarosanchez — CHANGES_REQUESTED** [review]

File: inject/src/main/java/io/micronaut/context/env/AbstractPropertySourceLoader.java (hunk 3 above)
Comment (altro3) [reply inject/src/main/java/io/micronaut/context/env/AbstractPropertySourceLoader.java:175]: fixed

File: inject/src/main/java/io/micronaut/context/env/AbstractPropertySourceLoader.java (hunk 4 above)
Comment (altro3) [reply inject/src/main/java/io/micronaut/context/env/AbstractPropertySourceLoader.java:103]: @alvarosanchez And here I would argue: calling `LOG.isDebugEnabled()` and other `isXXXEnabled` methods makes sense if you have some calculations before calling the logging method. If you just make a method call without calculation, like `log.debug("My message {}", myParam);`, it makes no sense for you to check the logging level, because methods debug / info / error etc alredy have it inside. By the way, this is one of the reasons why logging formatting was invented with symbols `{}`. in this place you haven't any calculation for logging and don't need to check log level twice

File: inject/src/main/java/io/micronaut/context/env/AbstractPropertySourceLoader.java (hunk 2 above)
Comment (altro3) [reply inject/src/main/java/io/micronaut/context/env/AbstractPropertySourceLoader.java:164]: fixe

File: .gitattributes (hunk 1 above)
Comment (altro3) [reply .gitattributes]: @sdelamo reverted this file

**Review by alvarosanchez — APPROVED** [review]

**Review by sdelamo — APPROVED** [review]
